flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/SessionContext.java

```diff
@@ -128,6 +137,26 @@ public SessionState getSessionState() {
         return sessionState;
     }
 
+    public boolean isStatementSetState() {
+        return isStatementSetState;
+    }
+
+    public void setStatementSetState(boolean isStatementSetState) {
+        this.isStatementSetState = isStatementSetState;
+    }
+
+    public List<ModifyOperation> getStatementSetOperations() {
+        return Collections.unmodifiableList(new ArrayList<>(statementSetOperations));
```

Comment:
I think 
```
        return Collections.unmodifiableList(statementSetOperations);
```

is enough.

Reply:
This is because that disableStatementSetMode() will clear the statementSetOperations, so here should return a new list.